fdb-extensions/src/main/java/com/apple/foundationdb/async/MoreAsyncUtil.java

```diff
@@ -836,4 +861,15 @@ public Boolean apply(T t) {
      * This is a static class, and should not be instantiated.
      **/
     private MoreAsyncUtil() {}
+
+    /**
+     * Exception that will be thrown when the <code>supplier</code> in {@link #getWithDeadline(long, Supplier)} fails to
+     * complete within the specified deadline time.
+     */
+    @SuppressWarnings("serial")
+    public static class DeadlineExceededException extends LoggableException {
+        private DeadlineExceededException() {
```

Comment:
It might be useful to have this take the deadline milliseconds as an argument and attach it to itself as a logging key/value pair so there is some kind of record of what the deadline was.

Reply:
Also, should this be a `RecordCoreException`?

Reply:
`MoreAsyncUtil` is in the extensions subproject while `RecordCoreException` is in the `fdb-record-layer-core` subproject. We could move `RecordCoreException` or leave as is, what would you prefer?